File: src/test/java/regressionTests/BankStatementTest.java
package regressionTests;

import com.cuupa.classificator.services.Classificator;
import com.cuupa.classificator.services.kb.semantic.SemanticResult;
import org.apache.pdfbox.pdmodel.PDDocument;
import org.apache.pdfbox.text.PDFTextStripper;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;
import regressionTests.config.TestConfig;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Objects;
import java.util.stream.Collectors;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

/**
 * @author Simon Thiel (https://github.com/cuupa)
 */
@SpringBootTest(classes = TestConfig.class)
@ActiveProfiles("test")
@RunWith(SpringRunner.class)
public class BankStatementTest extends RegressionTest {

    @Autowired
    private Classificator classificator;

    @Value("${test.paths.bank_statement}")
    private String path;

    @Test
    public void pathShouldNotBeEmpty() {
        assertNotNull(path);
    }

    @Test
    public void shouldBeBankStatement() throws Exception {
        List<List<SemanticResult>> semanticResults = callSemantik(path);

        for (List<SemanticResult> results : semanticResults) {
            boolean found = false;
            for (SemanticResult result : results) {
                if (result.getTopicName().equals("BANK_STATEMENT")) {
                    found = true;
                }
                System.out.println(result.getTopicName());
            }
            assertTrue(found);
        }
    }
}

File: src/test/java/regressionTests/BankStatementTest.kt
package regressionTests

import com.cuupa.classificator.services.Classificator
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.junit4.SpringRunner
import regressionTests.config.TestConfig

/**
 * @author Simon Thiel (https://github.com/cuupa)
 */
@SpringBootTest(classes = [TestConfig::class])
@ActiveProfiles("test")
@RunWith(SpringRunner::class)
class BankStatementTest : RegressionTest() {
    @Autowired
    private val classificator: Classificator? = null
    @Value("\${test.paths.bank_statement}")
    private val path: String? = null

    @Test
    fun pathShouldNotBeEmpty() {
        Assert.assertNotNull(path)
    }

    @Test
    @Throws(Exception::class)
    fun shouldBeBankStatement() {
        val semanticResults = callSemantik(path!!)
        for (results in semanticResults) {
            var found = false
            for ((topicName) in results) {
                if (topicName == "BANK_STATEMENT") {
                    found = true
                }
                println(topicName)
            }
            Assert.assertTrue(found)
        }
    }
}